.github/workflows/review.yml

```diff
@@ -69,7 +69,31 @@ on:
         default: false
 
 jobs:
+  prepare:
+    runs-on: ubuntu-latest
+    outputs:
+      pr: ${{ steps.pr.outputs.pr }}
+      head: ${{ steps.pr.outputs.head }}
+      merge: ${{ steps.pr.outputs.merge }}
+
+    steps:
+      - name: get pr
+        id: pr
+        run: |
+          pr="$(gh api "/repos/NixOS/nixpkgs/pulls/$PR_NUMBER")"
+          jq <<< "$pr"
+          if [[ "$(jq '.mergeable' <<< "$pr")" != "true" ]]; then
+            echo "::error::PR is not mergeable"
+            exit 1
+          fi
+          echo "pr=$pr" >> "$GITHUB_OUTPUT"
+          echo "head=$(jq -r '.head.sha' <<< "$pr")" >> "$GITHUB_OUTPUT"
+          echo "merge=$(jq -r '.merge_commit_sha' <<< "$pr")" >> "$GITHUB_OUTPUT"
+        env:
+          GH_TOKEN: ${{ github.token }}
+
   review:
+    needs: [prepare]
```

Comment:
Having a `prepare` job could also be used in future to simplify how the job matrix works.

Many of the things that need to be checked repeatedly using conditions could be baked directly into the matrix using an `includes` list.

To be fair, this could also be hard-coded as-is, without the `prepare` job.

Reply:
Yes, I will probably do that later